### PR TITLE
Paymaster tutorial: matches with repo code

### PR DESCRIPTION
**Pull request by uF4No**
Updates to match with repo code in [this PR](https://github.com/matter-labs/custom-paymaster-tutorial/pull/12)
Adds some clarifications

### CODE REVIEW

File: docs/dev/tutorials/custom-paymaster-tutorial.md [hunk 1]
```diff
@@ -257,19 +262,19 @@ contract MyPaymaster is IPaymaster {
             msg.sender == BOOTLOADER_FORMAL_ADDRESS,
             "Only bootloader can call this method"
         );
-        // Continue execution if called from the bootloader.
+        // Continure execution if called from the bootloader.
```
Comment (kailash-manasarovar) [inline docs/dev/tutorials/custom-paymaster-tutorial.md:265]:
```suggestion
        // Continue execution if called from the bootloader.
```

File: docs/dev/tutorials/custom-paymaster-tutorial.md [hunk 2]
```diff
@@ -56,32 +56,38 @@ mkdir contracts deploy
...
   defaultNetwork: "zkSyncTestnet",
-
   networks: {
+    hardhat: {
+      zksync: false,
+    },
     zkSyncTestnet: {
-      url: "https://testnet.era.zksync.dev",
-      ethNetwork: "goerli", // the RPC URL of the network (e.g. `https://goerli.infura.io/v3/<API_KEY>`)
+      url: "https://zksync2-testnet.zksync.dev",
```
Comment (kailash-manasarovar) [inline docs/dev/tutorials/custom-paymaster-tutorial.md:76]: ?

File: docs/dev/tutorials/custom-paymaster-tutorial.md [hunk 3]
```diff
@@ -257,19 +262,19 @@ contract MyPaymaster is IPaymaster {
...
         allowedToken = _erc20;
     }
 
-    function validateAndPayForPaymasterTransaction (
+    function validateAndPayForPaymasterTransaction(
         bytes32,
         bytes32,
         Transaction calldata _transaction
-    ) onlyBootloader external payable returns (bytes4 magic, bytes memory context) {
+    ) external payable returns (bytes4 magic, bytes memory context) {
```
Comment (kailash-manasarovar) [inline docs/dev/tutorials/custom-paymaster-tutorial.md:277]: ?

File: docs/dev/tutorials/custom-paymaster-tutorial.md [hunk 4]
```diff
@@ -328,20 +333,23 @@ contract MyPaymaster is IPaymaster {
...
 
-    function postTransaction  (
+    function postTransaction(
         bytes calldata _context,
         Transaction calldata _transaction,
         bytes32,
         bytes32,
         ExecutionResult _txResult,
         uint256 _maxRefundedGas
-    ) onlyBootloader external payable override {
```
Comment (kailash-manasarovar) [inline docs/dev/tutorials/custom-paymaster-tutorial.md:344]: ?

File: docs/dev/tutorials/custom-paymaster-tutorial.md [hunk 5]
```diff
@@ -358,7 +366,7 @@ Create the `contracts/MyERC20.sol` file and copy/paste the following:
 ```solidity
 // SPDX-License-Identifier: UNLICENSED
 
-pragma solidity ^0.8.8;
+pragma solidity ^0.8.0;
```
Comment (kailash-manasarovar) [inline docs/dev/tutorials/custom-paymaster-tutorial.md:369]: ?

**Review by kailash-manasarovar — CHANGES_REQUESTED** [review]
I'm not sure what I'm looking at. It looks like a bit of a mix of a previous version plus updates.

I wonder if you assumed the latest version was the repo and did the updates there first then copy/pasted, except there had been a bunch of updates in the doc since the repo was updated previously?

File: docs/dev/tutorials/custom-paymaster-tutorial.md [hunk 6]
```diff
@@ -236,13 +242,12 @@ Create the `contracts/MyPaymaster.sol` file and copy/paste the following:
 
 ```solidity
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.8;
+pragma solidity ^0.8.0;
```
[the remark this answers is not in the document]
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md:245]: we just need "^0.8.0" in the contracts as that will take any newer versions

File: docs/dev/tutorials/custom-paymaster-tutorial.md (hunk 1 above)
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md:265]: fix incoming

File: docs/dev/tutorials/custom-paymaster-tutorial.md (hunk 3 above)
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md:277]: fix incoming

File: docs/dev/tutorials/custom-paymaster-tutorial.md (hunk 4 above)
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md:344]: fix incoming

File: docs/dev/tutorials/custom-paymaster-tutorial.md (hunk 5 above)
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md:369]: we just need 0.8 as mentioned

File: docs/dev/tutorials/custom-paymaster-tutorial.md (hunk 4 above)
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md]: fixed in https://github.com/matter-labs/zksync-web-era-docs/pull/465/commits/3ae8c68a33e059a1bf7fdf6f4a84ae2185722510

File: docs/dev/tutorials/custom-paymaster-tutorial.md (hunk 3 above)
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md]: fixed in https://github.com/matter-labs/zksync-web-era-docs/pull/465/commits/3ae8c68a33e059a1bf7fdf6f4a84ae2185722510

File: docs/dev/tutorials/custom-paymaster-tutorial.md (hunk 2 above)
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md]: fixed in https://github.com/matter-labs/zksync-web-era-docs/pull/465/commits/3ae8c68a33e059a1bf7fdf6f4a84ae2185722510

File: docs/dev/tutorials/custom-paymaster-tutorial.md (hunk 1 above)
Comment (uF4No) [reply docs/dev/tutorials/custom-paymaster-tutorial.md]: fixed in https://github.com/matter-labs/zksync-web-era-docs/pull/465/commits/3ae8c68a33e059a1bf7fdf6f4a84ae2185722510

**Comment by uF4No** [comment]
@kailash-manasarovar all aligned now and re-tested with bootloader modifier

File: docs/dev/tutorials/custom-paymaster-tutorial.md [hunk 7]
```diff
@@ -265,11 +270,16 @@ contract MyPaymaster is IPaymaster {
...
-    function validateAndPayForPaymasterTransaction (
+    function validateAndPayForPaymasterTransaction(
         bytes32,
         bytes32,
         Transaction calldata _transaction
-    ) onlyBootloader external payable returns (bytes4 magic, bytes memory context) {
+    )
+        external
+        payable
+        onlyBootloader
```
Comment (kailash-manasarovar) [inline docs/dev/tutorials/custom-paymaster-tutorial.md:280]: I believe (could be wrong) the modifier has to go at the beginning of this list.

**Review by kailash-manasarovar — APPROVED** [review]
LGTM

**Pull request merged**